### ISSUE TITLE
EXCEPTION: Attempt to use a dehydrated detector.

### ISSUE THREAD

**Issue by sambhav-gore** [akserg/ng2-dnd#14]
I am getting this exception. 
It is most likely my mistake of how I configured everything. But have you seen this before ? 
I am trying to update an array when the drop is done and same component uses that array to display the lists (it is a calendar kind of component).

**Comment by akserg** [comment]
May I see the example code on Plunker or here? 
I saw an exception of dehydrated detector before but better to look at the stack trace.

**Comment by sambhav-gore** [comment]
This got resolved.
Problem was as follows:

I had a bunch of drop targets generated using ngFor.
There was a condition for not showing the drop target (after something was dropped on it) and I was using *ngIf for that. 

As explained in this accepted answer - http://stackoverflow.com/questions/35515254/what-is-a-dehydrated-detector-and-how-am-i-using-one-here ngIf was causing the issue.

I changed the condition to [hidden] instead of ngIf and issue got resolved.
